### ISSUE TITLE
TTL on Redis Retained Messages

### ISSUE THREAD

**Issue by akrizs** [rmqtt/rmqtt#138]
It is possible that i'm unable to find the option to set the TTL on the retained messages when using Redis as a backend for the `RMQTT-RETAINER` plugin.

but is it possible to ensure that the messages are retained until deleted, they are currently set to 5 minutes by default when the messages are stored into Redis and i just can't seem to figure out where to set the TTL, I have attempted to set the Message Expiry Interval on messages that are sent via the MQTTx client, but that does not seem to change anything.

I would need to ensure that the messages are persisted on Redis without TTL.

Best Wishes!

**Comment by bittcrafter** [comment]
Yes, currently the expiration time of retained messages is consistent with the expiration time of the messages themselves. I’ll think about whether to configure a separate TTL for retained messages.

**Comment by akrizs** [comment]
Yeah because a retained message should be able to be retained, i.e. if 0 would be set then the message would not timeout and would be retained until an empty message would be sent, have never seen an mqtt server that timeouts retained messages.

**Comment by bittcrafter** [comment]
Thank you for your suggestion, and we also discovered a bug. The current implementation allows retained messages to expire along with their message expiry interval by default, which is in compliance with the MQTT 5.0 specification. However, we provide a configurable option to modify the retained message expiry interval or to disable expiry altogether.



---- Replied Message ----
| From | Anton ***@***.***> |
| Date | 12/15/2024 20:21 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [rmqtt/rmqtt] TTL on Redis Retained Messages (Issue #138) |

Yeah because a retained message should be able to be retained, i.e. if 0 would be set then the message would not timeout and would be retained until an empty message would be sent, have never seen an mqtt server that timeouts retained messages.
